tests/template_tests/test_context.py

```diff
@@ -282,3 +282,14 @@ def test_modify_context_and_render(self):
         context = RequestContext(request, {})
         context['foo'] = 'foo'
         self.assertEqual(template.render(context), 'foo')
+
+    def test_context_render(self):
+        """
+        #27258 -- Raise an exception if RequestContext is used
+                with template.backends.django.Template.render()
+        """
+        template = Template('{{ foo }}')
```

Comment:
The test doesn't work because it doesn't run the code that's changed. This is `django.template.base.Template` and the modified function is called by `django.templates.backends.django.Template`.  I think the old assertion was passing because it was `self.assertRaises(TypeError, template.render(context))` but should have been `self.assertRaises(TypeError, template.render, context)`